src/suggestion-box.ts

```diff
@@ -0,0 +1,383 @@
+    private input_temp: HTMLElement | null | string = null;
+
+    connectedCallback() {
+        super.connectedCallback();
+        this.addEventListener("mouseout", this._mouseOutEvent);
+        document.addEventListener("click", this._outClickListener);
+    }
+    disconnectedCallback() {
+        super.disconnectedCallback();
+        this.removeEventListener("mouseout", this._mouseOutEvent);
+        document.addEventListener("click", this._outClickListener);
+    }
+    render() {
+        return html`
+            <div style="position:relative">
+                <div
+                    part="sgn-input"
+                    class="inputbox"
+                    @focusin=${this._inputFocusEvent}
+                    @keydown=${this.keyDownInputEvent}
+                >
+                    <slot
+                        name="sgn-input"
+                        @input=${this.InputEvent}
+                        @click=${this.inputfield}
+                        ><input type="search"
+                    /></slot>
+                </div>
+                <slot name="sgn-item-template" style="display:none"></slot>
+                <div
+                    part="sgn-container"
+                    style="display:${this.show_container ? "block" : "none"};"
+                >
+                    <div style="display:${this.show_empty ? "block" : "none"}">
+                        <slot name="sgn-empty"></slot>
+                    </div>
+                    <div
+                        style="display:${!this.show_empty ? "block" : "none"};"
+                        @click=${this.saveInputEvent}
+                        @mouseover=${this.mouseOverEvent}
+                    >
+                        <slot name="__private-item"></slot>
+                    </div>
+                </div>
+            </div>
+        `;
+    }
+    private _indexOfTarget(event: any) {
+        let el = event.target;
+        while (el) {
+            if (el.slot == "__private-item") break;
+            el = el.parentElement;
+        }
+        return this.private_items.indexOf(el);
+    }
+    saveInputEvent(event: Event) {
+        if (this.result[this._indexOfTarget(event)]) {
+            (this.input_el as HTMLInputElement).value =
+                this.result[this._indexOfTarget(event)]?.["name"];
+        } else if (this._indexOfTarget(event) === this.result.length) {
+            this.loadmoreentry(event);
+        }
+    }
+
+    mouseOverEvent(e: any) {
+        this.curr_index = this._indexOfTarget(e);
+        this.index = this.curr_index;
+
+        if (this.prevIndex >= 0) {
+            const prevItem = this.private_items[this.prevIndex];
+            if (prevItem) {
+                prevItem.classList.remove("sgn-active");
+            }
+        }
+
+        if (this.curr_index >= 0) {
+            this.private_items[this.curr_index].classList.add("sgn-active");
+        }
+
+        this.prevIndex = this.curr_index;
```

Comment:
This can be shrinked to 
`this.private_items[this.index]?.classList.remove('sgn-active'); 
this.index = this._indexOfTarget(e);
this.private_items[this.index]?.classList.add('sgn-active')`